Framework/CurveFitting/src/Algorithms/LeBailFunction.cpp

```diff
@@ -78,33 +79,32 @@ LeBailFunction::LeBailFunction(std::string peaktype) {
 
 //----------------------------------------------------------------------------------------------
 /** Destructor
-  */
+*/
 LeBailFunction::~LeBailFunction() {}
 
 //----------------------------------------------------------------------------------------------
 /** Return the composite function
-  */
+*/
 API::IFunction_sptr LeBailFunction::getFunction() {
   return m_compsiteFunction;
   // return boost::dynamic_pointer_cast<IFunction_sptr>(m_compsiteFunction);
 }
 
 //----------------------------------------------------------------------------------------------
 /** Calculate powder diffraction pattern by Le Bail algorithm
-  * @param out :: output vector
-  * @param xvalues :: input vector
-  * @param calpeaks :: if true, calculate peaks
-  * @param calbkgd :: if true, then calculate background and add to output.
- * otherwise, assume zero background
-  */
-void LeBailFunction::function(std::vector<double> &out,
-                              const std::vector<double> &xvalues, bool calpeaks,
-                              bool calbkgd) const {
-  if (out.size() != xvalues.size())
-    throw runtime_error("xvalues and out have different sizes.");
+* @param xvalues :: input vector
+* @param calpeaks :: if true, calculate peaks
+* @param calbkgd :: if true, then calculate background and add to output.
+* otherwise, assume zero background
+* @return :: output vector
+*/
+std::vector<double>
```

Comment:
As mentioned above, consider returning `HistogramY`, unless that does not work somewhere?